packages/react/src/components/createInlineOverlayComponent.tsx

```diff
@@ -107,13 +114,38 @@ export const createInlineOverlayComponent = <PropType, ElementType>(
           this.setState({ isOpen: false });
         }
 
-        this.props.onDidDismiss && this.props.onDidDismiss(evt);
+        const didComponentUnmount = this.ref.current === null;
+
+        if (didComponentUnmount === false) {
+          /**
+           * If the component is unmounted, we do not
+           * want to call the onDidDismiss handler to
+           * prevent a React state update on an unmounted component.
+           */
+          this.props.onDidDismiss && this.props.onDidDismiss(evt);
```

Comment:
Can we add a comment that explains why we don't remove the event listener in the on unmount method?